Islandora-CLAW/islandora
# Media source update

**GitHub Issue**: https://github.com/Islandora-CLAW/CLAW/issues/787

# What does this Pull Request do?

Provides an endpoint for you to PUT binaries and update files associated with Media.

# What's new?

* New controller
* New service for business logic required by the controller
* Updated routing and services yml files
* Functional tests

# How should this be tested?

Create a media and give it a file.  Let's say it's created at `/media/1`.  Then PUT a new file to `/media/1/source` like so: `curl -u admin:islandora -v -X PUT -H 'Content-Type: image/png' -H 'Content-Disposition: attachment; filename="my_image.png"' --data-binary @my_image.png localhost:8000/media/1/source`

* The source file should get updated
* Any fields extracted from the file should get updated as well (mimetype, image width/height, etc...)

Some things to note:
- Requests without a `Content-Type` header are rejected with 400.
- Requests without a `Content-Disposition` header are rejected with 400.
- Requests without a file name in the `Content-Disposition` header are rejected with 400.
- Requests without a body are rejected with 400.
- Requests from users without the 'update media' permission are rejected with 403.

# Interested parties
@Islandora-CLAW/committers @mjordan @rosiel
